__tests__/app.test.js

```diff
@@ -79,12 +79,81 @@ describe("GET: /api/articles/:article_id", () => {
             .expect(400)
             .then((  {body} )=> {
                 expect(body.msg).toBe("Invalid input, no such end point exists");
             })
     })
+})
+
+describe("PATCH: /api/articles/:article_id", () => {
+    describe("HAPPY path", () => {
+        it("Status 200: responds with a 200 to indicate that the patch was succesful and responds with the updated article.", () => {
+            const incVotes = {
+                inc_votes : 100
+            };
+            return request(app)
+                .patch('/api/articles/2')
+                .send(incVotes)
+                .expect(200)
+                .then(( {body} ) => {
+                    const { article } = body
+                    expect(article).toBeInstanceOf(Object);
+
+                    expect(article).toEqual(
+                        expect.objectContaining({
+                            article_id: expect.any(Number),
+                            author: expect.any(String),
+                            title: expect.any(String),
+                            topic: expect.any(String),
+                            body: expect.any(String),
+                            created_at: expect.any(String),
+                            votes: expect.any(Number),
+                        })
+                    )
+                    expect(article.votes).toBe(100)
+                })
+        });
+    })
+    describe("SAD path", () => {
+        it("Status 400: responds with a 400 and an appropriate message when a bad end point is provided", () => {
+            const incVotes = {
+                inc_votes : 100
+            };
+            return request(app)
+                .patch('/api/articles/banana')
+                .send(incVotes)
+                .expect(400) 
+                .then(( {body} )=> {
+                    expect(body.msg).toBe("Invalid input");
+                })
+        })
+        it("Status 400: responds with a 400 and an appropriate message when a valid end point is provided, but the body is unacceptable", () => {
+            const incVotes = {
+                inc_votes : "banana"
+            };
+            return request(app)
+                .patch('/api/articles/1')
+                .send(incVotes)
+                .expect(400) 
+                .then(( {body} )=> {
+                    expect(body.msg).toBe("Invalid input");
+                })
+        })
+        it("Status 404: responds with a 404 and an appropriate message when a valid enp point is provided but the resource doesn't exist", () => {
+            const incVotes = {
+                inc_votes : 100
+            };
+            return request(app)
+                .patch('/api/articles/1000')
+                .send(incVotes)
+                .expect(404)
+                .then(( {body} )=> {
+                    expect(body.msg).toBe("No article found for article_id 1000, patch unsuccessful");
+                })
+        })
+    })
```

Comment:
400 - missing required fields e.g.  incVotes = {}

Reply:
Thank you! Will update now!